pkg/server/cloud/aws/instances.go

```diff
@@ -267,17 +283,20 @@ func bootImageSpecToDescribeImagesInput(spec cloud.BootImageSpec) *ec2.DescribeI
 	return input
 }
 
-func getRootDeviceVolumeSizeAndType(blockDevices []*ec2.BlockDeviceMapping, rootDeviceName string) (int32, string) {
-	var rootDiskSize int32
-	volumeType := "gp2"
+func getRootDeviceVolumeSpecs(blockDevices []*ec2.BlockDeviceMapping, rootDeviceName string) EbsSpecs {
+	specs := EbsSpecs{VolumeType: "gp2"}
 	for _, blockDevice := range blockDevices {
 		if aws.StringValue(blockDevice.DeviceName) == rootDeviceName && blockDevice.Ebs != nil {
-			rootDiskSize = int32(aws.Int64Value(blockDevice.Ebs.VolumeSize))
-			volumeType = aws.StringValue(blockDevice.Ebs.VolumeType)
+			specs.VolumeSize = int32(aws.Int64Value(blockDevice.Ebs.VolumeSize))
+			specs.VolumeType = aws.StringValue(blockDevice.Ebs.VolumeType)
+			specs.Iops = aws.Int64Value(blockDevice.Ebs.Iops)
+			if specs.VolumeType == "gp3" {
+				specs.Throughput = aws.Int64Value(blockDevice.Ebs.Throughput)
```

Comment:
The AWS docs only say that this is only a valid parameter for gp3, but they don't mention if it is optional. I'd also make this one a pointer, and only set it in `getBlockDeviceMapping()` if it's non-nil here.